src/cmd/commit.test.ts

```diff
@@ -0,0 +1,103 @@
+import { AppsV1Api } from '@kubernetes/client-node'
+import { jest } from '@jest/globals'
+import { terminal } from '../common/debug'
+import { isOAuth2ProxyRunning } from './commit'
+import { mock } from 'jest-mock-extended'
+
+jest.mock('../common/debug')
+
+describe('isOAuth2ProxyRunning', () => {
+  const mockAppsV1Api = mock<AppsV1Api>()
+  const mockTerminal = terminal as jest.MockedFunction<typeof terminal>
+
+  const mockTerminalInfo = jest.fn()
+
+  beforeEach(() => {
+    mockTerminal.mockReturnValue({
+      info: mockTerminalInfo,
+    } as any)
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('should throw an error if the OAuth2 Proxy deployment is not found', async () => {
+    // @ts-ignore
+    mockAppsV1Api.readNamespacedDeployment.mockResolvedValue({ body: null })
+
```

Comment:
I noticed a couple of `// @ts-ignore` comments in the code, and I was thinking it might be better to use `as any` instead in these cases. It's generally preferable to avoid ignoring errors, as it helps keep the code more predictable.

Alternatively, if there are multiple `// @ts-ignore` comments, we could use `// @ts-nocheck` at the top of the file to make the code cleaner and easier to read.

Also, instead of repeatedly using `// eslint-disable-next-line @typescript-eslint/unbound-method`, we could place `/* eslint-disable @typescript-eslint/unbound-method */ `at the top of the file to simplify things.